src/libraries/Microsoft.Extensions.Logging.Abstractions/gen/LoggerMessageGenerator.Emitter.cs

```diff
@@ -606,16 +608,28 @@ private static string ConvertEndOfLineAndQuotationCharactersToEscapeForm(string
 
             return sb.ToString();
         }
-
-        private static bool ContainsAtSymbol(string value) => value.Length > 0 && value[0] == '@';
-
-        private static string ProtectAtSymbol(string value) =>
-            ContainsAtSymbol(value) ? value : $"_{value}";
-
-        private static (string template, string parameter) SanitizeAtSign(string template, string parameter)
-        {
-            static string SanitizeSingle(string input) => input.Length > 0 && input[0] == '@' ? input.Substring(1) : input;
-            return (SanitizeSingle(template), SanitizeSingle(parameter));
-        }
+        /// <summary>
+        /// Checks if string contains a special symbol ('@') as starting char
+        /// </summary>
+        /// <param name="input">true if contains special symbol, otherwise false.</param>
+        /// <returns></returns>
+        private static bool ContainsSpecialSymbol(ReadOnlySpan<char> input)
+            => input.Length > 0 && input[0] == '@';
+
+        /// <summary>
+        /// Removes '@' from input, allowing local variables to be declared when creating code.
+        /// </summary>
+        /// <param name="input">input that might contain '@' symbol</param>
+        /// <returns>If input starts with a symbol ('@'), returns a new string starting with underscore with '@' symbol removed, otherwise returns current string.</returns>
+        private static ReadOnlySpan<char> ProtectSpecialSymbolInVariable(ReadOnlySpan<char> input) =>
```

Comment:
>ProtectSpecialSymbolInVariable [](http://example.com/codeflow?start=42&length=30)

I suggest renaming this method to something like `NormalizeSymbolName` or something similar. The term `Protect` is not clear here.